java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -0,0 +1,330 @@
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+
+public class MainActivity 
+	extends ListActivity 
+	implements HealthVaultInitializationHandler {
+
+	private HealthVaultApp service;
+	private HealthVaultClient hvClient;
+	private BottomNavigationView mBottomNav;
+	private LinearLayout weightTile;
+
+	private ProgressDialog connectProgressDialog;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_main);
+		
+		service = HealthVaultApp.getInstance(this);
+		if (service.isAppConnected()) {
+			service.start(this, this);
+		}
+
+		hvClient = new HealthVaultClient();
+		
+		mBottomNav = (BottomNavigationView) findViewById(R.id.navigation);
+
+		mBottomNav.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
+			@Override
+			public boolean onNavigationItemSelected(MenuItem item) {
+				// handle desired action here
+				// One possibility of action is to replace the contents above the nav bar
+				// return true if you want the item to be displayed as the selected item
+				return true;
+			}
+		});
+
+		weightTile = (LinearLayout) findViewById(R.id.WeightTile);
+		weightTile.setOnClickListener(new View.OnClickListener() {
+			@Override
+			public void onClick(View view) {
+				Intent intent = null;
+				if (service.isAppConnected()) {
+					intent = new Intent(MainActivity.this, WeightActivity.class);
+				}
+				else
+				{
+					Toast.makeText(MainActivity.this, "Please connect to HV from Setting menu!", Toast.LENGTH_SHORT).show();
+				}
+				if(intent != null) {
+					startActivity(intent);
+				}
+			}
+		});
+	}
+	
+	@Override
+	protected void onStart() {
+		super.onStart();
+		hvClient.start();
+	}
+	
+	@Override
+    protected void onResume()
+    {
+		super.onResume();
+    }
+	
+    @Override
+	protected void onStop() {
+		super.onStop();
+		if(connectProgressDialog != null) {
+			connectProgressDialog.dismiss();
+		}
+		
+		hvClient.stop();
+	}
+    
+    @Override
+	public boolean onCreateOptionsMenu(Menu menu) {
+		getMenuInflater().inflate(R.menu.main, menu);
+		RelativeLayout recordNameLayout = (RelativeLayout) menu.findItem(R.id.record_name_layout).getActionView();
+        TextView tv = (TextView) recordNameLayout.findViewById(R.id.current_record_name);
+        
+        HealthVaultApp application = HealthVaultApp.getInstance();
+		if(service.isAppConnected() && application.getCurrentRecord() != null) {	
+	       tv.setText(application.getCurrentRecord().getName());
+	       
+	       recordNameLayout.setOnClickListener(new OnClickListener() {
+				
+				@Override
+				public void onClick(View arg0) {
+					Intent intent = new Intent(MainActivity.this, RecordPickerActivity.class);
+					startActivity(intent);
+				}
+			});
+		}
+		else {
+			tv.setVisibility(View.GONE);
+		}
+		
+		return true;
+	}
+
+	public void onConnected() {
```

Comment:
This is needed for override abstract method onConnected() in HealthVaultInitializationHandler

---
In reply to: [112363621](https://github.com/Microsoft/healthvault-samples/pull/4#discussion_r112363621) [](ancestors = 112363621)